packages/commonwealth/client/scripts/views/pages/CommunityManagement/CommunityProfile/Profile.tsx

```diff
@@ -1,7 +1,43 @@
 import React from 'react';
+import app from 'state';
+import Permissions from 'utils/Permissions';
+import FeatureHint from 'views/components/FeatureHint';
+import { CWText } from 'views/components/component_kit/cw_text';
+import { PageNotFound } from '../../404';
+import Form from './Form';
+import './Profile.scss';
 
 const CommunityProfile = () => {
-  return <div className="CommunityProfile">CommunityProfile</div>;
+  if (
+    !app?.user?.activeAccount?.address ||
+    !(Permissions.isSiteAdmin() || Permissions.isCommunityAdmin())
+  ) {
+    return <PageNotFound />;
+  }
```

Comment:
I was thinking of a HOC pattern or a custom hook, something like `withAdminCreds() HOC` or a `useCredentials() hook`, it will abstract this logic away. What do you think, what should we move ahead with.

Reply:
for sure not HoC as it is not up-to-date pattern. Definitely custom hook over the HoC. But as I mentioned before, maybe it could be handled on the route level?

Reply:
yeah, I agree we can do it there. I tried this diff and it seems to work, but since the Layout page is very easy to break so I want to do this in a dedicated PR. Thoughts?

```diff
diff --git a/packages/commonwealth/client/scripts/navigation/CommonDomainRoutes.tsx b/packages/commonwealth/client/scripts/navigation/CommonDomainRoutes.tsx
index bcd2c9946..c1541f97d 100644
--- a/packages/commonwealth/client/scripts/navigation/CommonDomainRoutes.tsx
+++ b/packages/commonwealth/client/scripts/navigation/CommonDomainRoutes.tsx
@@ -419,6 +419,7 @@ const CommonDomainRoutes = () => [
           path="/:scope/manage/profile"
           element={withLayout(CommunityProfile, {
             scoped: true,
+            isAdminOnly: true,
           })}
         />,
         <Route
@@ -426,6 +427,7 @@ const CommonDomainRoutes = () => [
           path="/:scope/manage/integrations"
           element={withLayout(CommunityIntegrations, {
             scoped: true,
+            isAdminOnly: true,
           })}
         />,
         <Route
@@ -433,6 +435,7 @@ const CommonDomainRoutes = () => [
           path="/:scope/manage/topics"
           element={withLayout(CommunityTopics, {
             scoped: true,
+            isAdminOnly: true,
           })}
         />,
         <Route
@@ -440,6 +443,7 @@ const CommonDomainRoutes = () => [
           path="/:scope/manage/moderators"
           element={withLayout(CommunityAdminAndModerators, {
             scoped: true,
+            isAdminOnly: true,
           })}
         />,
       ]
diff --git a/packages/commonwealth/client/scripts/views/Layout.tsx b/packages/commonwealth/client/scripts/views/Layout.tsx
index 1ef46edaa..d023c8376 100644
--- a/packages/commonwealth/client/scripts/views/Layout.tsx
+++ b/packages/commonwealth/client/scripts/views/Layout.tsx
@@ -12,14 +12,16 @@ import SubLayout from './Sublayout';
 import { CWEmptyState } from './components/component_kit/cw_empty_state';
 import { CWSpinner } from './components/component_kit/cw_spinner';
 import { CWText } from './components/component_kit/cw_text';
+import Permissions from '../utils/Permissions';
 
 type LayoutAttrs = {
   Component: ReactNode | any;
   scoped?: boolean;
+  isAdminOnly?: boolean;
   type:
-    | 'community' // Community-scoped layout with a header, CW sidebar, & community sidebar.
-    | 'common' // Generic layout with header and CW sidebar, used for non-community-scoped pages.
-    | 'blank'; //  Blank layout using Layout.scss styles
+  | 'community' // Community-scoped layout with a header, CW sidebar, & community sidebar.
+  | 'common' // Generic layout with header and CW sidebar, used for non-community-scoped pages.
+  | 'blank'; //  Blank layout using Layout.scss styles
 };
 
 // For complete documentation, see the App-Initialization-Flow.md wiki entry.
@@ -27,6 +29,7 @@ const LayoutComponent = ({
   Component, // Child component being rendered
   scoped = false,
   type = 'community',
+  isAdminOnly = false,
 }: LayoutAttrs) => {
   const routerParams = useParams();
   const pathScope = routerParams?.scope?.toString() || app.customDomainId();
@@ -107,9 +110,13 @@ const LayoutComponent = ({
 
     // If attempting to navigate to a community not fetched by the /status query, return a 404
     const pageNotFound = selectedScope && !scopeMatchesChain;
+    const isPageAccessible = isAdminOnly
+      ? app.isLoggedIn() && (Permissions.isCommunityAdmin() || Permissions.isSiteAdmin())
+      : true
+
     return (
       <Suspense fallback={Bobber}>
-        {pageNotFound ? <PageNotFound /> : <Component {...routerParams} />}
+        {(pageNotFound || !isPageAccessible) ? <PageNotFound /> : <Component {...routerParams} />}
       </Suspense>
     );
   };
```

Reply:
thanks!

Reply:
as discussed async, we will be doing it in a follow-up PR.